esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -2069,37 +2066,38 @@ void WsWuInfo::readWorkunitLog(IFileIOStream* ios, MemoryBuffer& buf, const char
     while (!eof)
     {
         eof = lineReader->readLine(line.clear());
-        if (strstr(line, startwuid))
-        {
-            outputThisLine = true;
-            outputALine(line.length(), line.str(), buf, outIOS);
-            processID = readLogProcessID(line);
-        }
-        else if (outputThisLine)
+        if (outputThisLine)
         {
             //If the slave is restarted before WU is finished, we cannot find out the "Finished wuid=...".
             //So, we should check whether the slave is restarting or not.
-            unsigned pID = readLogProcessID(line);
+            unsigned pID = 0;
+            bool foundEndWUID = parseLogLine(line, endwuid, pID);
             if ((pID > 0) && (pID != processID))
                 break;
 
             outputALine(line.length(), line.str(), buf, outIOS);
+            if (foundEndWUID)
+                outputThisLine = false;
+        }
+        else if (strstr(line, startwuid))
+        {
+            outputThisLine = true;
+            outputALine(line.length(), line.str(), buf, outIOS);
+            if (processID == 0)
+                parseLogLine(line, nullptr, processID);
         }
-
-        if (strstr(line, endwuid))
-            break;
     }
 }
 
-unsigned WsWuInfo::readLogProcessID(const char* line)
+bool WsWuInfo::parseLogLine(const char* line, const char* endWUID, unsigned& processID)
```

Comment:
I think it's worth adding a comment here to say what the expected format is.

This functionality will break, if an extra field is added into the logging ahead of the PID.
Really, you should be able to query the logging system to discover the format and then ask it to parse out desired fields - but that's for another time, but a comment would be good.

Reply:
Will add the comment.